### PR TITLE
Tests for Verifier Impls

### PR DESCRIPTION
**Pull request by claravanstaden**
Resolves: [SNO-562](https://linear.app/snowfork/issue/SNO-562)

### CODE REVIEW

File: parachain/pallets/ethereum-beacon-client/src/impls.rs [hunk 1]
```diff
@@ -71,7 +73,7 @@ impl<T: Config> Verifier for Pallet<T> {
 
 impl<T: Config> Pallet<T> {
 	/// Verifies that the receipt encoded in `proof.data` is included in the block given by
-	/// `proof.block_hash`. Inclusion is only recognized if the block has been finalized.
```
Comment (claravanstaden) [inline parachain/pallets/ethereum-beacon-client/src/impls.rs:74]: I think this isn't true on the `verify_receipt_inclusion` method anymore. Finalization checks are handled by the beacon client.

File: parachain/pallets/ethereum-beacon-client/src/tests.rs [hunk 2]
```diff
@@ -418,6 +417,27 @@ fn submit_update_with_sync_committee_in_current_period() {
...
+		assert_ok!(EthereumBeaconClient::submit(
+			RuntimeOrigin::signed(1),
+			sync_committee_update.clone()
+		));
+		assert_ok!(EthereumBeaconClient::submit(RuntimeOrigin::signed(1), update.clone()));
+		let block_root: H256 = update.finalized_header.clone().hash_tree_root().unwrap();
+		assert!(<FinalizedBeaconState<Test>>::contains_key(block_root));
+	});
+}
+
```
Comment (claravanstaden) [inline parachain/pallets/ethereum-beacon-client/src/tests.rs:440]: Accidentally removed this test in https://github.com/Snowfork/snowbridge/pull/894/files. Adding it back.

**Review by vgeddes — APPROVED** [review]
+1

**Review by alistair-singh — APPROVED** [review]

**Review by yrong — APPROVED** [review]
LGTM!

**Pull request merged**